### PR TITLE
[Support new order funnel - Part 3] Miscellaneous fixes for Dedicated Datacenter

### PR DESCRIPTION
**Pull request by FredericEspiau**
# Support for new order funnel
This PR is part of the **Support new order funnel** project
## Part 3 - Miscellaneous fixes for Dedicated Datacenter
It looks like there is a lot of changes but every commit contains only a few character changes.
If you remove whitespaces while reviewing, then you will see that not a lot have changed.
### :nail_care: Refactor
* Most controllers have been turned into classes
    * I know I could have done more and removed the `$scope` when possible and turn everything into a `Component`, but I didn't have the time, I just wanted to work on the files without ESlint annyoing me :)
### :bug: Bugfix
* The `name naming` dialogs will display a message and close properly in case of error
* Fixed some cosmetic issues

## References
- MANAGER-2372

### CODE REVIEW

File: client/app/dedicatedCloud/datacenter/host/dedicatedCloud-datacenter-host.routes.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 angular
   .module('App')
-  .config(($stateProvider) => {
+  .config(/* @ngInject */ ($stateProvider) => {
```
Comment (marie-j) [inline client/app/dedicatedCloud/datacenter/host/dedicatedCloud-datacenter-host.routes.js]: You could've added them to `controllers` too :)

File: client/app/dedicatedCloud/datacenter/backup/enable/dedicatedCloud-datacenter-backup-enable.controller.js [hunk 2]
```diff
@@ -26,19 +26,26 @@ angular
...
-        this.hosts = data.hosts;
-      }).finally(() => {
-        this.loading.init = false;
-      });
+      return this.$q
+        .all({
+          datacenter: this.DedicatedCloud
+            .getDatacenterInfoProxy(this.$stateParams.productId, this.$stateParams.datacenterId),
+          hosts: this.DedicatedCloud
+            .getHostsLexi(this.$stateParams.productId, this.$stateParams.datacenterId),
```
Comment (marie-j) [inline client/app/dedicatedCloud/datacenter/backup/enable/dedicatedCloud-datacenter-backup-enable.controller.js]: I know you've not written this part but maybe `getHostsLexi` should be renamed as it refers to legacy naming

File: client/app/dedicatedCloud/datacenter/host/dedicatedCloud-datacenter-host.routes.js (hunk 1 above)
Comment (FredericEspiau) [reply client/app/dedicatedCloud/datacenter/host/dedicatedCloud-datacenter-host.routes.js]: It's actually been done for most controllers !

https://github.com/ovh-ux/ovh-manager-dedicated/pull/1022/commits/25b20b765a609931e23f28d2be5b771c85c8eaa2

File: client/app/dedicatedCloud/datacenter/backup/enable/dedicatedCloud-datacenter-backup-enable.controller.js (hunk 2 above)
Comment (FredericEspiau) [reply client/app/dedicatedCloud/datacenter/backup/enable/dedicatedCloud-datacenter-backup-enable.controller.js]: I agree with destroying all mentions to `Lexi`

Comment (FredericEspiau) [reply client/app/dedicatedCloud/datacenter/backup/enable/dedicatedCloud-datacenter-backup-enable.controller.js]: It's been done

File: client/app/dedicatedCloud/datacenter/host/dedicatedCloud-datacenter-host.controller.js [hunk 3]
```diff
@@ -1,50 +1,83 @@
...
+              meta: {
+                totalCount: result.count,
+              },
+            }));
+        });
+    }
+
+    orderHost(datacenter) {
+      if (this.constants.target === 'US') {
+        this.$state.go('app.dedicatedClouds.datacenter.hosts.orderUS');
```
Comment (JeremyDec) [inline client/app/dedicatedCloud/datacenter/host/dedicatedCloud-datacenter-host.controller.js]:
```suggestion
        return this.$state.go('app.dedicatedClouds.datacenter.hosts.orderUS');
```

File: client/app/dedicatedCloud/datacenter/host/orderUS/dedicatedCloud-datacenter-host-orderUS.controller.js [hunk 4]
```diff
@@ -1,95 +1,123 @@
...
+
+    fetchInitialData() {
+      this.loading = true;
+
+      return this.$q
+        .all({
+          url: this.User.getUrlOf('express_order'),
+          user: this.User.getUser(),
+        })
+        .then((results) => {
```
Comment (JeremyDec) [inline client/app/dedicatedCloud/datacenter/host/orderUS/dedicatedCloud-datacenter-host-orderUS.controller.js]:
```suggestion
        .then(({ url, user }) => {
```

**Comment by Jisay** [comment]
@JeremyDec: your comments will be taken in consideration in part 4 ;-)

File: client/app/dedicatedCloud/datacenter/dedicatedCloud-datacenter.controller.js [hunk 5]
```diff
@@ -1,183 +1,221 @@
...
+            .copy(this.$scope.datacenter.model.description);
+          this.$scope.datacenterName.model = angular.copy(this.$scope.datacenter.model.name);
+          this.$scope.loadingInformations = false;
+        })
+        .catch((data) => {
+          this.$scope.loadingInformations = false;
+          this.$scope.loadingError = true;
+          this.$scope.setMessage(this.$translate.instant('dedicatedCloud_dashboard_loading_error'), angular.extend(data, { type: 'ERROR' }));
+        })
+        .then(() => this.DedicatedCloud
```
[the remark this answers is not in the document]
Comment (FredericEspiau) [reply client/app/dedicatedCloud/datacenter/dedicatedCloud-datacenter.controller.js]: The code before was
```js
Promise1()
  .catch(() => soSomething())

Promise2()
```

`Promise2` executes even if `Promise1` fails

To have the same comportement, I have either to `$q.all`, either to 
```js
Promise1()
  .catch(() => soSomething())
  .then(() => Promise2())
```

I don't remember why I didn't `$q.all`, the order of execution must be important

File: client/app/dedicatedCloud/datacenter/host/dedicatedCloud-datacenter-host.controller.js (hunk 3 above)
Comment (FredericEspiau) [reply client/app/dedicatedCloud/datacenter/host/dedicatedCloud-datacenter-host.controller.js]: I won't do it as the issue is there about 20 times and each times it means returning importing `$q` and returning an empty Promise 😞

File: client/app/dedicatedCloud/datacenter/dedicatedCloud-datacenter.controller.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (JeremyDec) [reply client/app/dedicatedCloud/datacenter/dedicatedCloud-datacenter.controller.js]: Yeah I guess so, forget this comment then :p

**Pull request merged**